plugin/interface_generator/main.go

```diff
@@ -141,6 +141,33 @@ func FieldListDestruct(structPrefix string, fieldList *ast.FieldList, fileset *t
 	return strings.Join(result, ", ")
 }
 
+func FieldListToRecordSuccess(structPrefix string, fieldList *ast.FieldList, fileset *token.FileSet) string {
+	if fieldList == nil || len(fieldList.List) == 0 {
+		return ""
+	}
+
+	result := ""
+	nextLetter := 'A'
+	for _, field := range fieldList.List {
+		typeNameBuffer := &bytes.Buffer{}
+		err := printer.Fprint(typeNameBuffer, fileset, field.Type)
+		if err != nil {
+			panic(err)
+		}
+		typeName := typeNameBuffer.String()
```

Comment:
This is not really required. You can get the base type from `field.Type` through some recursion:

```go
func baseTypeName(x ast.Expr) string {
	switch t := x.(type) {
	case *ast.Ident:
		return t.Name
	case *ast.SelectorExpr:
		if _, ok := t.X.(*ast.Ident); ok {
			// only possible for qualified type names;
			// assume type is imported
			return t.Sel.Name
		}
	case *ast.ParenExpr, *ast.StarExpr:
		return baseTypeName(t.X)
	}
	return ""
}
```

And then use `baseTypeName(typeName) == "error" || baseTypeName(typeName) == "*model.AppError"`. I haven't tested this though. This is just code copied from the standard library.

Reply:
Doesn't it reduce readability of code?

Reply:
IMO, this is the right way to get the type of an expression. What we are doing right now is a bit hackish, by printing the expression and then comparing the string.

It's just code which might be a bit difficult to understand at first, but it doesn't look unreadable to me.

Reply:
Oh. I didn't know that. I was just following the way it was done in other functions.

So should I change them all to reflect it this way?

Reply:
Oh my indeed ! I didn't notice that there are other places where we are doing this. There's no need to change other stuff in this PR. But maybe add this function and use it in this PR, and then send another PR migrating the other functions ?

Thanks.

Reply:
Could you share the link to where this code is, in the standard library?

Currently this throws error
```
❯ make pluginapi
go generate  ./plugin
# command-line-arguments
interface_generator/main.go:207:24: t.X undefined (type ast.Expr has no field or method X)
plugin/client_rpc.go:4: running "go": exit status 2
make: *** [Makefile:241: pluginapi] Error 1
```

Reply:
That is strange, all of those expressions have `X` in their field.

It is from here:
https://github.com/golang/go/blob/2a4e71c752875f84dc0061f75fdcbfc96ae8d5f2/src/go/doc/reader.go#L100-L116